gateway-provider-rewrite/src/main/java/org/apache/knox/gateway/filter/rewrite/impl/json/JsonUrlRewriteFilterReader.java

```diff
@@ -52,14 +52,19 @@ public JsonUrlRewriteFilterReader(
 
   @Override
   protected String filterValueString(String name, String value, String rule ) {
-    try {
-      Template input = Parser.parseLiteral( value );
-      Template output = rewriter.rewrite( resolver, input, direction, rule );
-      value = output.getPattern();
-    } catch( URISyntaxException e ) {
-      LOG.failedToParseValueForUrlRewrite( value );
+    if(rule != null || UrlRewriter.Direction.OUT == direction) {
```

Comment:
Might help to add a big comment here about what and why.

Reply:
sure, good idea !